src/main/java/science/icebreaker/controller/WikiController.java

```diff
@@ -53,6 +60,19 @@ public int addWikiPage(
                     .withErrorCode(ErrorCodeEnum.ERR_WIKI_001);
         }
 
+        // Check if the referenced keywords exist
+        Set<String> existingKeywords
+                = networkService.getAllKeywordNodes().stream().map(Node::getName).collect(Collectors.toSet());
+        if (wikiPage.getNetworkKeywords() != null) {
+            for (String networkKeyword : wikiPage.getNetworkKeywords()) {
+                if (!existingKeywords.contains(networkKeyword)) {
+                    throw new IllegalRequestParameterException()
```

Comment:
I would prefer to collect all missing keywords and throwing an error contain all.